File: modules/iam/src/main/java/com/media/bus/iam/member/controller/MemberController.java
package com.media.bus.iam.member.controller;

import com.media.bus.common.result.type.CommonResult;
import com.media.bus.common.web.response.DataView;
import com.media.bus.iam.member.dto.FindMemberByEmailRequest;
import com.media.bus.iam.member.dto.FindMemberByJwtRequest;
import com.media.bus.iam.member.dto.FindMemberByLoginIdRequest;
import com.media.bus.iam.member.dto.FindMemberByMemberIdRequest;
import com.media.bus.iam.member.dto.MemberResponse;
import com.media.bus.iam.member.service.MemberService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@Tag(name = "유저(Member) API", description = "회원조회, 회원 수정 등 회원 관련 API")
@RestController
@RequestMapping("/api/v1/member")
@RequiredArgsConstructor
public class MemberController {

    private final MemberService memberService;

    /// 로그인시 사용된 jwt 토큰으로 회원 조회.
    @Operation(summary = "회원조회", description = "jtw를 기준으로 회원을 조회합니다.")
    @PostMapping("/jwt")
    public DataView<MemberResponse> findByJwtMember(@RequestBody @Valid FindMemberByJwtRequest request) {
        return DataView.<MemberResponse>builder()
            .result(CommonResult.SUCCESS)
            .data(memberService.findByJwtMember(request.jwt()))
            .build();
    }

    /// 유저 아이디를 통한 회원 조회.
    @Operation(summary = "회원조회", description = "memberId를 기준으로 회원을 조회합니다.")
    @PostMapping("/id")
    public DataView<MemberResponse> findByMemberId(@RequestBody @Valid FindMemberByMemberIdRequest request) {
        return DataView.<MemberResponse>builder()
            .result(CommonResult.SUCCESS)
            .data(memberService.findByMemberId(request.memberId()))
            .build();
    }

    /// 유저 아이디를 통한 회원 조회.
    @Operation(summary = "회원조회", description = "memberId를 기준으로 회원을 조회합니다.")
    @PostMapping("/login-id")
    public DataView<MemberResponse> findByLoginId(@RequestBody @Valid FindMemberByLoginIdRequest request) {
        return DataView.<MemberResponse>builder()
            .result(CommonResult.SUCCESS)
            .data(memberService.findByLoginId(request.loginId()))
            .build();
    }

    /// 유저 이메일을 통한 회원 조회.
    @Operation(summary = "회원조회", description = "memberId를 기준으로 회원을 조회합니다.")
    @PostMapping("/email")
    public DataView<MemberResponse> findByEmail(@RequestBody @Valid FindMemberByEmailRequest request) {
        return DataView.<MemberResponse>builder()
            .result(CommonResult.SUCCESS)
            .data(memberService.findByEmail(request.email()))
            .build();
    }

}

File: modules/iam/src/main/kotlin/com/media/bus/iam/member/controller/MemberController.kt
package com.media.bus.iam.member.controller

import com.media.bus.common.web.response.ApiResponse
import com.media.bus.contract.security.annotation.Authorize
import com.media.bus.iam.member.dto.FindMeRequest
import com.media.bus.iam.member.dto.MemberModifyRequest
import com.media.bus.iam.member.dto.MemberResponse
import com.media.bus.iam.member.service.MemberService
import io.swagger.v3.oas.annotations.Operation
import io.swagger.v3.oas.annotations.tags.Tag
import jakarta.validation.Valid
import org.springframework.web.bind.annotation.*

@Tag(name = "유저(Member) API", description = "External 전용 회원조회, 회원 수정 등 회원 관련 API")
@RestController
@RequestMapping("/api/v1/member")
class MemberController(
    private val memberService: MemberService,
) {

    /**
     * 이름으로 아이디 찾기
     */
    @Operation(summary = "아이디 찾기", description = "회원 이름, 전화번호, email로 아이디를 검색합니다.")
    @GetMapping("/find/me")
    fun findMe(@RequestBody @Valid request: FindMeRequest): ApiResponse<MemberResponse> =
        ApiResponse.success(memberService.findMe(request))


    /**
     * 회원 정보 수정
     * 2차 본인 인증 완료 후에만 호출 가능하다.
     * 회원 정보 수정 후 2차 인증 상태를 삭제하여 1회성으로 사용한다.
     */
    @Authorize
    @Operation(summary = "회원 정보 수정", description = "회원 정보를 수정합니다.")
    @PostMapping("/modify")
    fun modify(@RequestHeader("X-User-Id") memberId: String, @RequestBody request: MemberModifyRequest): ApiResponse<Unit?> =
        ApiResponse.success(memberService.modify(memberId, request))


    /**
     * 회원 탈퇴.
     * 2차 본인 인증 완료 후에만 호출 가능하다.
     * 탈퇴 처리 완료 후 2차 인증 상태를 삭제하여 1회성으로 사용한다.
     */
    @Authorize
    @Operation(summary = "회원 탈퇴", description = "2차 본인 인증 완료 후 회원 탈퇴를 처리합니다.")
    @DeleteMapping("/withdraw")
    fun withdraw(@RequestHeader("X-User-Id") memberId: String): ApiResponse<Unit?> =
        ApiResponse.successWith { memberService.withdraw(memberId) }
}